src/github.com/getlantern/flashlight/geolookup/geolookup.go

```diff
@@ -0,0 +1,94 @@
+package geolookup
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/getlantern/flashlight/ui"
+	"github.com/getlantern/geolookup"
+	"github.com/getlantern/golog"
+)
+
+const (
+	messageType = `GeoLookup`
+
+	sleepTime = time.Second * 10
+)
+
+var (
+	log = golog.LoggerFor("geolookup-flashlight")
+
+	service     *ui.Service
+	lookupMutex sync.Mutex
+	lookupData  *geolookup.City
+)
+
+func getUserGeolocationData() *geolookup.City {
+	lookupMutex.Lock()
+	defer lookupMutex.Unlock()
+
+	var err error
+
+	if lookupData != nil {
+		// We already looked up IP's information.
+		return lookupData
+	}
+
+	for {
+		if !geolookup.UsesDefaultHTTPClient() {
```

Comment:
Also, just because it's not using the default http client doesn't mean that it's using the client that we want :)

One way to make this cleaner would be to have the geolookup.LookupCity() function accept an HttpClient. If provided, it uses that, and if not it uses the default.  That way, we can be sure that it's using the client we want when we call it, and we don't have to bother managing the httpClient variable in geolookup.